news-bundle/src/EventListener/DataContainer/PageListener.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\NewsBundle\EventListener\DataContainer;
+
+use Contao\BackendUser;
+use Contao\CoreBundle\ServiceAnnotation\Callback;
+use Contao\CoreBundle\ServiceAnnotation\Hook;
+use Contao\NewsBundle\Controller\NewsFeedController;
+use Doctrine\DBAL\Connection;
+use Symfony\Component\Security\Core\Security;
+
+class PageListener
+{
+    public function __construct(private Connection $connection, private Security $security)
+    {
+    }
+
+    /**
+     * @Callback(table="tl_page", target="fields.archives.options")
+     */
+    public function getAllowedArchives(): array
+    {
+        $user = $this->security->getUser();
+
+        $stmt = $this->connection->createQueryBuilder()
+            ->select('id, title')
+            ->from('tl_news_archive')
+        ;
+
+        if (!($user instanceof BackendUser || $this->security->isGranted('ROLE_ADMIN'))) {
+            $stmt->where('id', $user->news);
```

Comment:
can we `->where()` on an array of IDs?

Reply:
Thanks for this comment – you are right of course. The implementation was actually completely bonkers. I fixed it and added basic unit tests in 718eab070